lib/elixir/test/elixir/map_test.exs

```diff
@@ -84,9 +84,16 @@ defmodule MapTest do
 
   test "update maps" do
     assert %{two_items_map | a: 3} == %{a: 3, b: 2}
-
-    assert_raise ArgumentError, fn ->
-      %{two_items_map | c: 3}
+    
+    case :erlang.system_info(:otp_release) do
+      '18' ->
+        assert_raise BadKeyException, fn ->
+          %{two_items_map | c: 3}
+        end
+      _ ->
```

Comment:
you should check for version '17' as that is the lowest supported version by elixir,
and anything else should be what you have in the first clause.